hackathon_site/dashboard/frontend/src/slices/event/teamSlice.ts

```diff
@@ -52,6 +58,84 @@ export const getCurrentTeam = createAsyncThunk<
     }
 });
 
+export const leaveTeam = createAsyncThunk<
+    Team,
+    void,
+    { state: RootState; rejectValue: RejectValue; dispatch: AppDispatch }
+>(
+    `${teamReducerName}/leaveTeam`,
+    async (_, { dispatch, getState, rejectWithValue }) => {
+        const teamCode = teamCodeSelector(getState());
+
+        try {
+            const response = await post<Team>("/api/event/teams/leave_team/", {
+                team_code: teamCode,
+            });
+
+            dispatch(
+                displaySnackbar({
+                    message: `You have left the team.`,
+                    options: { variant: "success" },
+                })
+            );
+
+            return response.data;
+        } catch (e: any) {
+            // order reached quantity limits
+            const errorData = e.response?.data?.non_field_errors;
```

Comment:
Note that this is an array, so we may need to handle displaying multiple errors. Alternatively you could always take the first error in the list (assume it's the most important) and display it only.

Reply:
can you elaborate more on this Yuying? I tried to console.log the error, but I didn't find any array.

<img width="494" alt="CleanShot 2022-08-21 at 17 00 28@2x" src="https://user-images.githubusercontent.com/36619969/185810764-1590667f-6e42-4b1f-bc2f-2a2c3070f483.png">

Reply:
I was probably thinking about something else, leave it as is. Could you remove the console log though